File: 1329-sort-the-matrix-diagonally/1329-sort-the-matrix-diagonally.cpp
class Solution {
public:
    vector<vector<int>> diagonalSort(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        unordered_map<int,priority_queue<int,vector<int>,greater<int>>>Map;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                Map[i-j].push(matrix[i][j]);
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                matrix[i][j] = Map[i-j].top();
                Map[i-j].pop();
            }
        }
        
        return matrix;
    }
};

File: 1329-sort-the-matrix-diagonally/1329-sort-the-matrix-diagonally.java
class Solution {
    public int[][] diagonalSort(int[][] mat) {
        int m = mat.length, n = mat[0].length;
    HashMap<Integer,PriorityQueue<Integer>>Map = new HashMap<>();
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                Map.putIfAbsent(i-j,new PriorityQueue<>());
                Map.get(i-j).add(mat[i][j]);
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                mat[i][j] = Map.get(i-j).peek();
                Map.get(i-j).poll();
            }
        }
        
        return mat;
    }
}